src/sentry/static/sentry/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import {Location} from 'history';
+
+import {Client} from 'app/api';
+import EventView, {
+  isAPIPayloadSimilar,
+  LocationQuery,
+} from 'app/utils/discover/eventView';
+import {EventQuery} from 'app/actionCreators/events';
+
+export type GenericChildrenProps<T> = {
+  isLoading: boolean;
+  error: null | string;
+  tableData: T | null;
+  pageLinks: null | string;
+};
+
+export type DiscoverQueryProps = {
+  api: Client;
+  /**
+   * Used as the default source for cursor values.
+   */
+  location: Location;
+  eventView: Pick<EventView, 'isValid' | 'getEventsAPIPayload'>;
```

Comment:
Why pick vs use the EventView entirely?

Reply:
I thought it would conflict if you needed to pass a modified shape of EventView, but just checked and I guess not 👍